docs/proposals/20220126_k1-for-devs.md

```diff
@@ -0,0 +1,76 @@
+# KubeOne for Developers
+
+## Goals
+* "Developer Batteries included"
+* Developers should get enabled to start working on the Kubernetes Cluster without having to deal with infrastructure setup problems
+* Independent KubeOne AddOns which can be enabled/disabled depending on the needs
+
+## Non Goals
+* Having the plugins available in KKP.
+* Setting up Development Environments for specific programming languages.
+* Being to opiniated about tooling. Eg zsh or fish should not be included.
+* Installing opiniated IDE AddOns. Only the minimal set should be delivered.
+
+## Motivation and Background
+For Developers it is still hard to get started with Kubernetes. For getting into production mode they have to care about some infrastructure tasks.
+
+For example:
+* Getting an IDE up and running
+* Connect to the Kubernetes Cluster
+* Installing the Kubernetes Dashboard
+* Having a Container Registry in Place
+* Having a proper Git setup
+* Having GitOps tooling up and running
+* Having kubectl and other useful tools (kubectx, bash completion, ) installed
+* Having a tool in place for handling sensitive data
+
+## Implementation
+
+### IDE
+* Using [Eclipse Che](https://github.com/eclipse/che) as Cloud IDE. Eclipse Che is a workspace server integrating Theia as a default. 
+* Theia is a nodejs application which provides the IDE. Workspace management has to be provided by Eclipse Che. 
```

Comment:
For the future, I think we should consider VS Code as the default but still need some development https://github.com/eclipse/che/issues/20500

Reply:
Yes we should definitely consider something different than Theia. I had bad experiences with it due to nodejs dependency hell.